lib/utils/get-proxy-agent.js

```diff
@@ -0,0 +1,15 @@
+const {getProxyForUrl} = require('proxy-from-env');
+const HttpsProxyAgent = require('https-proxy-agent');
+const NPM_REGISTRY = 'https://registry.npmjs.org/';
+
+let proxyAgent = false;
+
+module.exports = function getProxyAgent() {
+    // Initialize Proxy Agent for proxy support if needed
+    const proxyAddress = getProxyForUrl(NPM_REGISTRY);
+    if (proxyAddress) {
```

Comment:
like @vikaspotluri123 said, can we move the require for the `https-proxy-agent` dependency down inside this function? that way, the agent is only ever required if it's used, and since we cache the proxy agent itself in theory the require should only happen once.